### PR TITLE
feat: Migrate Legal Hold Modal to React (ACC-197)

### PR DESCRIPTION
**Pull request by przemvs**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-197" title="ACC-197" target="_blank">TaskACC-197</a>  Migrate legal-hold-modal to React Component
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .betterer.cache [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (atomrc) [inline .betterer.cache]: :thinking:  This should be ignored by our .gitignore

File: src/script/components/Modals/LegalHoldModal/LegalHoldModal.tsx [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+          )}
+        </div>
+      </ModalComponent>
+    </div>
+  );
+};
+
+export default LegalHoldModal;
+
+registerReactComponent('legal-hold-modal', LegalHoldModal);
```
Comment (atomrc) [inline src/script/components/Modals/LegalHoldModal/LegalHoldModal.tsx]: Can we use the modal helper rather that creating a new knockout component (cf. https://github.com/wireapp/wire-webapp/pull/13232/files#diff-f3ac9e91a8a405d63b806880efb60f009db63d1e1144c9c06ad604feedde4d1cR96)?

Would that work?

**Review by atomrc — CHANGES_REQUESTED** [review]
Can the LegalHoldViewModel be deleted?

File: src/script/components/Modals/LegalHoldModal/LegalHoldModal.tsx (hunk 2 above)
Comment (przemvs) [reply src/script/components/Modals/LegalHoldModal/LegalHoldModal.tsx]: We can't pass everything what's we need for legal hold modal using modal helper :(

File: .betterer.cache (hunk 1 above)
Comment (przemvs) [reply .betterer.cache]: Strange.. let me check

**Comment by przemvs** [comment]
> Can the LegalHoldViewModel be deleted?

Nah, it's used also in ew legal-hold-dot

**Comment by atomrc** [comment]
@przemvs Let's try our best to completely remove the `legalHoldModalViewModel`. This should be possible by
- creating a small API to trigger the modal (using the `renderModal` helper that we created)
- using this api to show the modal with the right param (like state and userId if needed)
- remove all calls to `amplify` 

Now we are dealing with 2 implementation of the legalhold modal, would be good to go all the way and migrate all the calls to the modal :)

**Comment by przemvs** [comment]
> @przemvs Let's try our best to completely remove the `legalHoldModalViewModel`. This should be possible by
> 
> * creating a small API to trigger the modal (using the `renderModal` helper that we created)
> * using this api to show the modal with the right param (like state and userId if needed)
> * remove all calls to `amplify`
> 
> Now we are dealing with 2 implementation of the legalhold modal, would be good to go all the way and migrate all the calls to the modal :)

I think the best idea to wait for merge all stuff from '<div id="app" class="app">', and then finish the LegalHoldModal, because we will create context for repositories, and then use it everywhere. Now we need to pass from parent > children > children, etc...

**Review by atomrc — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**